### ISSUE TITLE
Error:  #268: declaration may not appear after executable statement in block

### ISSUE THREAD

**Issue by suilin1254703825** [LekKit/sha256#2]
Under the scenario using LLVM for cross-compilation, **"Error: #268: Declaration may not Appear after Executable Statement in block"** will be prompted.  **It is recommended that variables be declared in advance at the beginning**.
For example:
``` c++
// old:
void sha256_read(const struct sha256_buff *buff, uint8_t *hash)
{
    for (uint32_t i = 0; i < 8; i++)
    {
        hash[i * 4] = (buff->h[i] >> 24) & 255;
        hash[i * 4 + 1] = (buff->h[i] >> 16) & 255;
        hash[i * 4 + 2] = (buff->h[i] >> 8) & 255;
        hash[i * 4 + 3] = buff->h[i] & 255;
    }
}

// new 
void sha256_read(const struct sha256_buff *buff, uint8_t *hash)
{
    uint32_t i = 0;
    for (/*uint32_t*/ i = 0; i < 8; i++)
    {
        hash[i * 4] = (buff->h[i] >> 24) & 255;
        hash[i * 4 + 1] = (buff->h[i] >> 16) & 255;
        hash[i * 4 + 2] = (buff->h[i] >> 8) & 255;
        hash[i * 4 + 3] = buff->h[i] & 255;
    }
}
```

**Comment by LekKit** [comment]
Thanks, I wasn't aware my code isn't C89 compliant (compilers that I've used for testing enable C99 by default, my bad), the issue is fixed in [8b1f77b](https://github.com/LekKit/sha256/commit/8b1f77bf7588cc63d4ae4677e2d97384f97089b3) - tested with -std=c89, but report if you have additional issues

**Comment by suilin1254703825** [comment]
I'm sorry,My label is not very accurate, thank you.
